src/nemo-devicelock/private/settingswatcher.h

```diff
@@ -81,6 +81,7 @@ class NEMODEVICELOCK_EXPORT SettingsWatcher : public QSocketNotifier, public QSh
     int showNotifications;
     int maximumAutomaticLocking;
     int absoluteMaximumAttempts;
+    qint64 temporaryLockTimeout;
```

Comment:
In addition to this we need the start time of the lock to be saved, which will be `ms` from epoch - I think the granularity here is enough. This way the lock can survive between boots if the temporary lock has been set.

Reply:
Start time stored in a data file elsewhere. In this context (= what value to show in settings app / stte) it does not matter.

Reply:
Inside the devicelock we're using deprecated toTime_t ( https://doc.qt.io/qt-5/qdatetime-obsolete.html#toTime_t ) we should be using toSecsSinceEpoch instead (https://doc.qt.io/qt-5/qdatetime.html#toSecsSinceEpoch). qint64 just aligns with that.

Changing those toTime_t to toSecsSinceEpoch is yet another thing and not related to this per se.

Reply:
We need to use `toMSecsSinceEpoch()` as others are in 6.4 only. 

And it would be straightforward to have all settings in one place but whatever.